### PR TITLE
Better error message for event deletion when event has attendances

### PR DESCRIPTION
**Pull request by echang594**
# Info

Closes **#199**.

# Description

What changes did you make? List all distinct problems that this PR addresses. Explain any relevant
motivation or context.

Throw error when trying to delete event with attendances.

## Changes

- Also fixed tests that assumed repository query results are always ordered correctly.

# Type of Change

- [x] Patch (non-breaking change/bugfix)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] Documentation (A change to a README/description)
- [ ] Continuous Integration/DevOps Change (Related to deployment steps, continuous integration
      workflows, linting, etc.)
- [ ] Other: (Fill In) 

If you've selected Patch, Minor, or Major as your change type, **make sure to bump the version before merging in `package.json`!**
# Testing

I have tested that my changes fully resolve the linked issue ...

- [x] locally.
- [ ] on the testing API/testing database.
- [x] with appropriate Postman routes. Screenshots are included below.

# Checklist

- [x] I have performed a self-review of my own code.
- [x] I have followed the style guidelines of this project.
- [x] I have appropriately edited the API version in the `package.json` file.
- [x] My changes produce no new warnings.

# Screenshots
Screenshot 2024-02-27 at 5 49 36 PM
The event uuid above is one that has attendances.

### CODE REVIEW

**Review by maxwn04 — APPROVED** [review]
🚀 LGTM 🚀

**Review by nik-dange — CHANGES_REQUESTED** [review]
Just left one comment to add a test case, other than that LGTM. Great work on your first feature, and awesome job on making the admin tests more concise! localeCompare to the rescue 😁

File: tests/event.test.ts [hunk 1]
```diff
@@ -126,3 +127,30 @@ describe('event creation', () => {
...
+    const eventController = ControllerFactory.event(conn);
+    await expect(eventController.deleteEvent({ uuid: event.uuid }, admin))
+      .rejects.toThrow('Cannot delete event that has attendances');
+
+    // verify event not deleted
+    const repository = conn.getRepository(EventModel);
+    const repositoryEvent = await repository.findOne({ uuid: event.uuid });
+
+    expect(repositoryEvent).toEqual(event);
+  });
```
[the remark this answers is not in the document]
Comment (nik-dange) [reply tests/event.test.ts:155]: Also, bump the patch number in package.json for this bug fix

[the remark this answers is not in the document]
Comment (echang594) [reply tests/event.test.ts:155]: Alright

**Review by nik-dange — APPROVED** [review]
LGTM 🚀

**Pull request merged**